code/modules/power/supermatter/supermatter.dm

```diff
@@ -1,438 +1,876 @@
+	var/warning_point = 50
+	///The alert we send when we've reached warning_point
+	var/warning_alert = "Danger! Crystal hyperstructure integrity faltering!"
+	///The point at which we start sending messages to the common channel
+	var/emergency_point = 700
+	///The alert we send when we've reached emergency_point
+	var/emergency_alert = "CRYSTAL DELAMINATION IMMINENT."
+	///The point at which we delam
+	var/explosion_point = 900
+	///When we pass this amount of damage we start shooting bolts
+	var/damage_penalty_point = 550
+
+	///A scaling value that affects the severity of explosions.
+	var/explosion_power = 35
+	///Time in 1/10th of seconds since the last sent warning
+	var/lastwarning = 0
+	///Refered to as eer on the moniter. This value effects gas output, heat, damage, and radiation.
+	var/power = 0
+	///Determines the rate of positve change in gas comp values
+	var/gas_change_rate = 0.05
+
+	var/n2comp = 0					// raw composition of each gas in the chamber, ranges from 0 to 1
+	var/plasmacomp = 0
+	var/o2comp = 0
+	var/co2comp = 0
+	var/n2ocomp = 0
+
+	///The last air sample's total molar count, will always be above or equal to 0
+	var/combined_gas = 0
+	///Affects the power gain the sm experiances from heat
+	var/gasmix_power_ratio = 0
+	///Affects the amount of o2 and plasma the sm outputs, along with the heat it makes.
+	var/dynamic_heat_modifier = 1
+	///Affects the amount of damage and minimum point at which the sm takes heat damage
+	var/dynamic_heat_resistance = 1
+	///Uses powerloss_dynamic_scaling and combined_gas to lessen the effects of our powerloss functions
+	var/powerloss_inhibitor = 1
+	///Based on co2 percentage, slowly moves between 0 and 1. We use it to calc the powerloss_inhibitor
+	var/powerloss_dynamic_scaling= 0
+	///Affects the amount of radiation the sm makes. We multiply this with power to find the rads.
+	var/power_transmission_bonus = 0
+	///Used to increase or lessen the amount of damage the sm takes from heat based on molar counts.
+	var/mole_heat_penalty = 0
+	///Takes the energy throwing things into the sm generates and slowly turns it into actual power
+	var/matter_power = 0
+	///How much the bullets damage should be multiplied by when it is added to the internal variables
+	var/bullet_energy = 2
+	///How much hallucination should we produce per unit of power?
+	var/hallucination_power = 0.1
+
+	///Our internal radio
+	var/obj/item/radio/radio
+
+	///Boolean used for logging if we've been powered
+	var/has_been_powered = FALSE
+	///Boolean used for logging if we've passed the emergency point
+	var/has_reached_emergency = FALSE
+
+	///An effect we show to admins and ghosts the percentage of delam we're at
+	var/obj/effect/countdown/supermatter/countdown
+
+	///Used along with a global var to track if we can give out the sm sliver stealing objective
+	var/is_main_engine = FALSE
+	///Our soundloop
+	var/datum/looping_sound/supermatter/soundloop
+	///Can it be moved?
+	var/moveable = FALSE
```

Comment:
Redundant by `anchored`?

Reply:
Nah, this is for if you want the SM to be anchorable or not---the engine, for example, can't be unanchored; the one you purchase from cargo, however, can.